rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+other component's APIs and data models because trying to get other component designs to change is just too difficult.
+These local sub-optimal choices keep piling up and introduce lots of unnecessary additional complexity to the architecture.
+
+An alternative approach is a radical reorganization of the DCP into theme teams drawn from annual roadmap
+objectives rather than per component group. A theme team is now responsible for their roadmap theme and the features
+that derive from it. They can make
+changes in any component. Each team is responsible for both their feature and the working of the DCP as a whole.
+
+There's no question there are considerable challenges with this approach. It requires more communication, which will be 
+a particular problem for the EBI to CZI and UCSC. I think this is surmountable - community open-source projects often have a 
+very wide distribution but can still co-ordinate decisions. Often this is through tools such as mailing lists rather 
+than real time, which will require a development culture change.
+
+There is also an argument that the knowledge required by feature teams, teams that cross component boundaries, is 
+simply too great - the DCP is just too complex to be understood by a single person. But I think that with the current
+DCP organization, complexity is high *because of* component boundaries rather than in spite of them. Not because
+partitioning complexity isn't a good idea, it absolutely is. Rather the alignment of institutions to components
+mean that the incentives (local camaraderie and reward) of sticking to your component are very high.
+ 
+A theme team that's considering the DCP as a whole may be able to make decisions
+that optimize across the whole of the DCP rather than for an individual component at the greater expense of others.
+
+In addition, theme teams have greater flexibility to adapt to changing user circumstances and implementation discoveries
+than per-component teams, especially if they are working to an indicator rather than detailed roadmap goals.
+
+Pros:
+1) Theme teams can consider implementations that optimize across the whole of the DCP, rather than being incentivized to
+optimize their local component at the expense of cross-DCP optimization.
+2) Theme teams are more flexible to adapt their work as circumstances emerge and change, rather than requiring a 
+planning cycle.
+3) It may be easier for developers to move between theme teams as required, rather than being tied to components that
+face varying workloads.
+4) It will be easier to create theme teams to work on cross-cutting DCP objectives, such as metrics of reliability or
+whole-DCP dashboards.
+5) More developers get a chance to lead if there are theme teams.
+6) Theme teams grow their familiarity with the DCP as a whole. There are no longer per-component silos, though one 
+would have to avoid per-theme silos emerging.
+
+Cons:
```

Comment:
Another Con: Institutions have (CZI) grants that call out component-specific work (e.g. UCSC will develop a Data Browser). Relating themes back to grant deliverables will be more challenging.

Reply:
This is a good point. Going forward, we will change the reporting to reflect progress of the DCP on the whole. It can be a single report, if each of the partners prefers this, that covers progress against the Roadmap and needs of the HCA. The goal has always been to create a single platform that meets the need of the HCA community and is flexible and well engineered. That is still very much the high level goal but recent work has clarified more specific aims and priorities. The goal of the roadmap is to help us all align to shared goals and priorities and avoid building (and reporting!) on fragmented pieces.

Reply:
@theathorn Yeah, that's an excellent point. Having the components embedded into the contracts considerably reduces our room for reorganization.